enterprise/server/execution_search_service/execution_search_service.go

```diff
@@ -53,6 +59,44 @@ type ExecutionWithInvocationId struct {
 	invocationID string
 }
 
+func (s *ExecutionSearchService) FetchExecutionRequestMetadata(ctx context.Context, invocationID string, execIDs []string) (map[string]*repb.RequestMetadata, error) {
+	if !*enableFetchExecutionRequestMetadata {
+		return make(map[string]*repb.RequestMetadata), nil
+	}
+	if s.oh == nil {
+		return nil, status.UnavailableError("OLAP DB is required to search execution request metadata")
+	}
+	u, err := s.env.GetAuthenticator().AuthenticatedUser(ctx)
+	if err != nil {
+		return nil, err
+	}
+	if u.GetGroupID() == "" {
+		return nil, status.InvalidArgumentError("failed to find user's group when searching executions request metadata.")
+	}
+	if err := authutil.AuthorizeGroupAccess(ctx, s.env, u.GetGroupID()); err != nil {
+		return nil, err
+	}
+	q := query_builder.NewQuery(`SELECT * FROM "Executions"`)
+	// Always filter to the currently selected (and authorized) group.
+	q.AddWhereClause("group_id = ?", u.GetGroupID())
+	q.AddWhereClause("invocation_uuid IN ?", invocationID)
+	q.AddWhereClause("execution_id IN ?", execIDs)
```

Comment:
maybe @luluz66 or @jdhollen can double-check my thinking on this - I think we probably need to filter by `updated_at_usec` to something like, invocation_start_time to invocation_end_time. Otherwise I think each query has to scan the full `invocation_id` column, since the sort key is `(group_id, updated_at_usec, invocation_id)`. The `invocation_id` column is one of the largest columns in the Executions table so I think it will be slow.

Reply:
+1 to filter by updated_at_usec

Reply:
I added start_time based on the first create_time of the mysql executions. However im not too sure about end_time. Technically we would only store clickhouse entries after the mysql entries, so do we use the highest MySQL updated_time plus some value (1 * time.Minute) for the upper bound? 🤔

Reply:
Executions are flushed to ClickHouse when the invocation is complete, which can happen a long time after the last execution is complete. If we aren't already looking up the invocation in this query, then I think it'd be fine to start with something like: the updated timestamp of the last mysql execution plus or minus 3 hours.